### PR TITLE
Rtos whdnet merge

### PR DESCRIPTION
**Pull request by jaenrig-ifx**
This is the result of merging these two branches
- https://github.com/jaenrig-ifx/micropython/tree/psoc6_network_jens
- https://github.com/jaenrig-ifx/micropython/tree/ports/psoc6_whdnet
after updating them with the psoc6 upstream contribution.

All test under github actions are now passing, and the basic RTOS setup is ready. Cleanup, refactoring and some dependency management will come next.

### CODE REVIEW

File: ports/psoc6/boards/CY8CPROTO-062-4343W/makefile_mtb.mk [hunk 1]
```diff
@@ -57,20 +57,23 @@ mpy_mtb_clean:
...
 #	$(info CFLAGS : $(CFLAGS))
 	$(eval CXXFLAGS += $(CFLAGS))
 #	$(info CXXFLAGS : $(CXXFLAGS))
 	$(eval LDFLAGS += -mcpu=cortex-m4 --specs=nano.specs -mfloat-abi=softfp -mfpu=fpv4-sp-d16 -mthumb -ffunction-sections -fdata-sections -ffat-lto-objects -g -Wall -pipe -Wl,--gc-sections -T$(MPY_DIR_OF_MTB_ADAPTER_MAKEFILE)/bsps/TARGET_APP_CY8CPROTO-062-4343W/COMPONENT_CM4/TOOLCHAIN_GCC_ARM/linker.ld -Wl,-Map,$(BUILD)/firmware.map -Wl,--start-group -Wl,--end-group -Wl,--print-memory-usage)
 #	$(info LDFLAGS : $(LDFLAGS))
-	$(eval QSTR_GEN_CFLAGS += $(INC))
+	$(eval QSTR_GEN_CFLAGS += $(INC) $(CFLAGS))
```
Comment (bargfred) [inline ports/psoc6/boards/CY8CPROTO-062-4343W/makefile_mtb.mk:76]: Cleanup : I think this should be moved into the main Makefile next to the other QSTR_GLOABL_DEPENDENCIES option as this is not related to MTB in any way.

**Review by bargfred — APPROVED** [review]
Great job !

File: ports/psoc6/boards/CY8CPROTO-062-4343W/makefile_mtb.mk (hunk 1 above)
Comment (jaenrig-ifx) [reply ports/psoc6/boards/CY8CPROTO-062-4343W/makefile_mtb.mk:76]: We can move it later together with the cleanup.

**Review by IFX-Anusha — APPROVED** [review]
Looks Good!
